### PR TITLE
This pull request is to merge rspec_tic_tac_toe branch to the Development branch

### PR DESCRIPTION
**Pull request by henatan99**
### **Testing Tic-tac-toe project using rspec.** 

- the first step of this project is installing rspec
- After installation rspec initialized in the project root directory 
- Test files are created for each class files: player_spec.rb, board_spec.rb and tictactoe_spec.rb, in the spec folder
- Positive and negative scenario examples are used for all the methods in the classes

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  describe 'update_board' do
+    it ' returns arg[1]' do
+      board1 = Board.new
+      expect(board1.update_board(2, 'X')).to eq('X')
+    end
+
+    # it " returns arg[1]" do
+    #     board1 = Board.new
+    #     expect(board1.update_board(2)).to raise_error(ArgumentError)
+    # end
```
Comment (peterrobert) [inline spec/board_spec.rb]: Commented code should be deleted. This will make your work clean.
messy

File: spec/player_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    end
+    # it "returns Argument error when object is attempted to be created by name only" do
+    #     expect(Player.new('Nikola')).to be_a(ArgumentError)
+    # end
+    # it "returns Argument error when a player object is attempted to be created by symbol only" do
+    #     expect(player3).to raise_error(ArgumentError)
+    # end
+    # it "returns Argument error when object is attempted to be created with no argument" do
+    #     expect(player4).to raise_error(ArgumentError)
+    # end
```
Comment (peterrobert) [inline spec/player_spec.rb]: Kindly remove all the commented code in your project

File: spec/player_spec.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+require './lib/player' #=> add this
+describe Player do
+  let(:player1) { Player.new('Henok', 'X') }
+  let(:player2) { Player.new('Nikola') }
+  let(:player3) { Player.new('X') }
+  let(:player4) { Player.new }
+  describe 'initialize' do
+    it 'returns player object pointer when a player object is created by name and symbol' do
+      expect(player1).to be_kind_of(Player)
+    end
```
Comment (peterrobert) [inline spec/player_spec.rb:11]: Kindly you should test for both positive and negative scenarios for instance:
`returns player object pointer when a player object is created by name and symbol` :+1:  But what if the player object doesn't have a name or a symbol. You should test for that failure too.

**Nb:** Kindly do so for all your tests. Make sure you check for both positive and negative scenarios.

commonsense

File: spec/tictactoe_spec.rb [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+      players = game.create_players('Henok', 'Nikola', 'X', 'O')
+      expect(players[0].sym).to eql('X')
+    end
+    it 'returns symbol of player two' do
+      players = game.create_players('Henok', 'Nikola', 'X', 'O')
+      expect(players[1].sym).to eql('O')
+    end
+    # it "returns Argument error when wrong number of argumets is given" do
+    #     expect(game.create_players('Henok', 'Nikola', 'X')).to raise_error(ArgumentError)
+    # end
```
Comment (peterrobert) [inline spec/tictactoe_spec.rb]: Kindly delete commented code

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED :recycle: :stop_sign: 
HELLO, @henatan99  AND @zdrale VERY NICE PROGRESS. I LOVE THAT YOU HAVE CREATED TESTS FOR ALL OF YOUR METHODS. JUST NEED SOME FEW CHANGES AND WE SHALL BE GOOD TO GO. KINDLY CHECK BELOW :arrow_down: 

- [ ] There are some rubocop errors in your project. Kindly fix all the errors. :arrow_down: 

rubocop

**summary**

- [ ] Kindly test for both the positive and negative scenarios. Just like I have described it below.

- [ ] Kindly remove all the commented code in your project.

- [ ] Your pull request message is descriptive. However, you should add a title and small description points telling us what you have worked on in this project

### [OPTIONAL]

- [ ] Using `require_relative` is usually in preference compared to just using `require` to link files in your project. Kindly try and use require_relative/


### if you have any question kindly reach me on slack @peterrobert


### HAPPY CODING :computer:

**Review by PhillipUg — APPROVED** [review]
# Status: Project Approved! 🥇 
Congratulations on implementing all project requirements and getting this project approved. :+1:

You can merge this branch into the development branch when you're ready and go on to your next project.
:bulb:  Do not forget to submit your project completion form after merging this branch.

All the best team,

For any assistance, you can contact me on slack as **Phillip Musiime**

**Pull request merged**